test/unittests/commands/xmlupload/stash/test_stash_circular_references_from_intermediary_resource.py

```diff
@@ -0,0 +1,187 @@
+
+
+@pytest.fixture
+def simple_text_value() -> IntermediarySimpleText:
+    return IntermediarySimpleText("Text", "prop", None, None)
+
+
+@pytest.fixture
+def link_value_to_res_1() -> IntermediaryLink:
+    return IntermediaryLink("res_1", "prop", None, None, "link_to_res_1_uuid")
+
+
+@pytest.fixture
+def link_value_with_permissions_to_res_2() -> IntermediaryLink:
+    return IntermediaryLink(
+        "res_2", "prop", None, Permissions({PermissionValue.CR: ["knora-admin:ProjectAdmin"]}), "link_to_res_2_uuid"
+    )
+
+
+@pytest.fixture
+def link_value_to_resource_no_links() -> IntermediaryLink:
+    return IntermediaryLink("res_no_links", "prop", None, None, str(uuid4()))
+
+
+@pytest.fixture
+def text_value_no_links() -> IntermediaryRichtext:
+    return IntermediaryRichtext(FormattedTextValue("No links"), "prop", None, None, set(), str(uuid4()))
+
+
+@pytest.fixture
+def text_value_with_link() -> IntermediaryRichtext:
+    return IntermediaryRichtext(
+        FormattedTextValue('Link: <a class="salsah-link" href="IRI:res_3:IRI">res_3</a>'),
+        "prop",
+        None,
+        None,
+        set("res_3"),
+        "standoff_link_to_res_3_uuid",
+    )
+
+
+@pytest.fixture
+def resource_1(
+    link_value_with_permissions_to_res_2: IntermediaryLink,
+    link_value_to_resource_no_links: IntermediaryLink,
+    simple_text_value: IntermediarySimpleText,
+) -> IntermediaryResource:
+    return IntermediaryResource(
+        res_id="res_1",
+        type_iri="type",
+        label="lbl",
+        permissions=None,
+        values=[link_value_with_permissions_to_res_2, link_value_to_resource_no_links, simple_text_value],
+    )
+
+
+@pytest.fixture
+def resource_2(text_value_with_link: IntermediaryRichtext) -> IntermediaryResource:
+    return IntermediaryResource(
+        res_id="res_2",
+        type_iri="type",
+        label="lbl",
+        permissions=None,
+        values=[text_value_with_link],
+    )
+
+
+@pytest.fixture
+def resource_3(link_value_to_res_1: IntermediaryRichtext) -> IntermediaryResource:
+    return IntermediaryResource(
+        res_id="res_3",
+        type_iri="type",
+        label="lbl",
+        permissions=None,
+        values=[link_value_to_res_1],
+    )
+
+
+@pytest.fixture
+def resource_no_links() -> IntermediaryResource:
+    return IntermediaryResource(
+        res_id="res_no_links",
+        type_iri="type",
+        label="lbl",
+        permissions=None,
+        values=[],
+    )
+
+
+def test_stash_circular_references_remove_link_value(
+    resource_1: IntermediaryResource,
+    resource_2: IntermediaryResource,
+    resource_3: IntermediaryResource,
+    resource_no_links: IntermediaryResource,
+) -> None:
+    copied_res_1 = deepcopy(resource_1)
+    copied_res_2 = deepcopy(resource_2)
+    copied_res_3 = deepcopy(resource_3)
+    resources = [copied_res_1, copied_res_2, copied_res_3, deepcopy(resource_no_links)]
+    lookup = {"res_1": ["link_to_res_2_uuid"]}
```

Comment:
Not part of this PR, but it struck my mind: When looking at this dict, it might be difficult to understand its purpose. One sees resource identifiers as keys, and then UUIDs as values. But what are the meaning of the UUIDs? 

I think it might be worth to make a class with some informative field names.